packages/patternfly-4/react-table/src/components/Table/Table.js

```diff
@@ -139,9 +139,29 @@ class Table extends React.Component {
   constructor(props) {
     super(props);
     this.state = {
-      headerData: []
+      headerData: [],
+      allRowsSelected: props.onSelect ? this.areAllRowsSelected(props.rows) : false
     };
+    this.isSelected = this.isSelected.bind(this);
+    this.areAllRowsSelected = this.areAllRowsSelected.bind(this);
   }
+
+  componentDidUpdate(prevProps) {
+    if (this.props.rows !== prevProps.rows) {
+      this.setState({
+        allRowsSelected: this.props.onSelect ? this.areAllRowsSelected(this.props.rows) : false
+      });
+    }
+  }
+
+  isSelected(row) {
+    return row.selected === true;
+  }
+
+  areAllRowsSelected(rows) {
+    return rows.every(this.isSelected);
```

Comment:
Should we skip expanded row children?

It's odd that both parent and child have an selected property, because they could be initialized differently and/or be out of sync. For example, setting the selected property on a child has no affect on its parent.

Reply:
Ok, I'll have to look into it, get up to speed on how the expand w/ children is working.

Reply:
If I weed them out myself, will rows.every(this.selected) still work?

Reply:
I don't think it's good idea to skip expanded children. There is option to show checkbox for expanded children as well, that might be handy in future once we figure out tree tables.

Reply:
Sounds reasonable. Thanks for the explanation